lib/truffle/truffle/patching.rb

```diff
@@ -0,0 +1,38 @@
+module Truffle::Patching
+  TRUFFLE_PATCHES_DIRECTORY = File.join(Truffle::Boot.ruby_home, 'lib', 'patches')
+  TRUFFLE_PATCHES           = Dir.glob(File.join(TRUFFLE_PATCHES_DIRECTORY, '**', '*.rb')).
+      select { |path| File.file? path }.
+      map { |path| path[(TRUFFLE_PATCHES_DIRECTORY.size+1)..-4] }.
+      reduce ({}) { |h, v| h.update v => true }
```

Comment:
It's evaluated only once over about a dozen of files, I don't think performance is relevant in this particular case. As for functional vs. imperative style I think this is matter of taste, I find functional clearer, more readable. Since I think it's only a matter of personal taste in this case, I'll leave it as is.